src/ReTestItems.jl

```diff
@@ -290,26 +290,23 @@ function _runtests_in_current_env(
             for (i, testitem) in enumerate(testitems.testitems)
                 testitem.workerid[] = Libc.getpid()
                 testitem.eval_number[] = i
-                nretries = 0
-                retry_limit = max(retries, testitem.retries)
-                while nretries ≤ retry_limit
+                run_number = 1
+                max_runs = 1 + max(retries, testitem.retries)
+                while run_number ≤ max_runs
                     res = runtestitem(testitem, ctx; verbose_results, logs)
                     ts = res.testset
-                    print_errors_and_captured_logs(testitem, nretries + 1; logs)
+                    print_errors_and_captured_logs(testitem, run_number; logs)
                     report_empty_testsets(testitem, ts)
-                    if ts.anynonpass && nretries < retry_limit
```

Comment:
unrelated, but `ts.anynonpass` is awful. every time I see it it makes me think someone is mispelling `anonymous`